zkEVM - add new reorg doc (#632)
0xPolygon/polygon-docs: docs/zkEVM/architecture/protocol/synchronizer-reorg.md

!!!info
    
    Currently, reorgs do not occur in the zkEVM system because a single sequencer is implemented, the _trusted sequencer_.

The possibility of L2 reorgs arises only in the event of L1 reorgs.

There’s a slight chance for a reorg occurrence in the event of [_forced batches_](../protocol/malfunction-resistance/sequencer-resistance.md), where a user is sequencing a batch.

However, the design of forced batches is specifically crafted to mitigate such scenarios.

## L1 reorgs

L1 reorgs happen if there is a reorg in Ethereum itself.

In general, L1 reorgs should never happen because once a state is consolidated, then users have the guarantee that the transactions are finalized.

L1 reorgs are far more critical, since it might be the case that the sequencer has to re-synchronize already virtualized or consolidated batches.

A reorg in L1 requires a change of the state.

The figure below, depicts a reorg scenario where two batches are in the virtual state and one batch is in the consolidated state.

![Figure: Reorg in L1 requires state change](../../../img/zkEVM/sync-reorg-in-l1-state.png)

> The synchronizer is in charge of detecting these types of scenario, and must inform the sequencer to perform the appropriate reorg.

Reviewer: ```suggestion
The synchronizer is responsible for identifying such scenarios and informing the sequencer to perform the appropriate reorg.
```
